lib/domain/data/user_data_holder.dart

```diff
@@ -1,4 +1,12 @@
 import 'package:flutter_dasher/common/model/user.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'user_data_holder.g.dart';
+
+@Riverpod(keepAlive: true)
+UserDataHolder userDataHolder(UserDataHolderRef ref) {
+  return UserDataHolder();
+}
 
 class UserDataHolder {
```

Comment:
Could have this just been, like a regular provide class?
```suggestion
@Riverpod(keepAlive: true)
class UserDataHolder {
```

Reply:
Sadly no. To use the `@riverpod` above the class, the class needs to be a `Notifier`:
```dart
@riverpod
class MyNotifier extends _$MyNotifier {
  @override
  Result build() {}
}
```

To provide regular classes object you must create a function:
```dart
@riverpod
RegularClass regularClass(regularClassRef ref) {
   return RegularClass();
}
```